### ISSUE TITLE
Theme Appleton: White space under footer area

### ISSUE THREAD

**Issue by mxhassani** [Automattic/wp-calypso#66552]
### Quick summary

There is a white space (added block) under the footer area of this theme.
Can be related to this bug: #59363

### Steps to reproduce

1. Atomic site
2. Activate the theme Appleton
3. Use the headstart home page
4. Select "hide" under Appearance → Customize → Site Identity → Footer Credit
5. Check the home page

### What you expected to happen

To see the full footer with a background color: 
![image](https://user-images.githubusercontent.com/81776920/184511507-262a376a-0671-41a4-b970-8e7cbf2beef2.png)

### What actually happened

I can see a white space under the footer:
![Uploading image.png…]()

### Context

5475646-zd-woothemes

### Platform (Simple, Atomic, or both?)

Atomic

### Theme-specific issue?

Looks like it - Appleton

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Workaround details

Can be done using CSS: 

```
/* Remove white space under footer |  */
footer.wp-block-template-part + .wp-block-group{
  display: none !important;
}
```

Replacing the template part with another one also solves the issue

**Comment by zachspears** [comment]
Theme has been updated to FSE, but I am still able to reproduce the issue. Here are the updated steps:
 
1. Atomic site
2. Activate the theme Appleton
3. Use the headstart home page
4. Select "hide" under Settings → General → Footer Credit → Footer Credit
5. Check the home page

Possible related: https://github.com/Automattic/wp-calypso/issues/59363

**Comment by jartes** [comment]
Another case in 37479826-hc after hiding the WordPress.com footer credit.

Thanks for the CSS Snippet!

**Comment by siraht** [comment]
I just ran into this issue as well. CSS snippet above fixes it.

**Comment by liviopv** [comment]
Sounds like the same issue as https://github.com/Automattic/themes/issues/6066

**Comment by devNigel** [comment]
Another case: 39582996-hc

This site is using Twenty Twenty-Three. 

Shared CSS fix.

**Comment by mgozdis** [comment]
Another case: 39694899-hc

Site is using Lettre theme. Shared CSS workaround.

**Comment by thekingsprojects** [comment]
5937148-hc

Heiwa theme. CSS workaround does the trick.

**Comment by danielbachhuber** [comment]
A seemingly similar report in p1675163387825869-slack-C04C3KCM42H

I have a fix in progress with https://github.com/Automattic/wpcomsh/pull/1245

**Comment by danielbachhuber** [comment]
Fixed for Simple sites too with D100793-code

One thing worth noting...

Some themes (like Zoologist) have the footer credit added as a block to the footer template:

![image](https://user-images.githubusercontent.com/36432/217819895-acf4e297-a440-4196-91ea-0f8fdf57af15.png)

When this is the case, our logic is only smart enough to strip out the credit string. It's not smart enough to traverse upwards and strip out the container. In this scenario, "Hiding the credit" requires also removing the corresponding group block.

However, when the footer credit doesn't exist as a block in the footer template, we programmatically append it:

![image](https://user-images.githubusercontent.com/36432/217820706-3ec832c8-2f74-44d1-a187-324eeef323d7.png)

For this scenario, my fix ensures the container HTML is removed when the credit link isn't displayed.
